src/lib/components/Communities/CommunityHeader.js

```diff
@@ -1,68 +1,84 @@
-  community,
-  imagePlaceholderLink,
-  setCommunity,
-}) => {
-  const communityLogo = community?.links?.logo
-    ? community.links.logo
-    : imagePlaceholderLink;
-  const communityTitle = community?.metadata?.title
-    ? community.metadata.title
-    : community;
+class CommunityHeaderComponent extends Component {
+  render() {
+    const { changeSelectedCommunity, community, imagePlaceholderLink } =
+      this.props;
 
-  return (
-    <Container className="deposits-community-header" fluid>
-      <Container className="community-header">
-        {community ? (
-          <Image
-            className="community-logo-header"
-            src={communityLogo}
-            fallbackSrc={imagePlaceholderLink}
-          />
-        ) : (
-          <span>{i18next.t('No community selected.')}</span>
-        )}
-
-        <div className="community-header-info">
-          {communityTitle}
-          <CommunitySelectionModal
-            onCommunityChange={setCommunity}
-            chosenCommunity={community}
-            trigger={
-              <Button
-                size="mini"
-                className="community-header-button ml-5"
-                color="blue"
-                name="setting"
-                type="button"
-              >
-                {i18next.t('Edit')}
-              </Button>
-            }
-          />
-        </div>
+    return (
+      <Container className="deposits-community-header" fluid>
+        <Container className="community-header">
+          {community ? (
+            <>
+              <Image
+                className="community-logo-header"
+                src={community.logo || imagePlaceholderLink}
+                fallbackSrc={imagePlaceholderLink}
+              />
+              {community.title || community.uuid}
+            </>
+          ) : (
+            <span>{i18next.t('No community selected.')}</span>
+          )}
+          <div className="community-header-info">
+            <CommunitySelectionModal
+              onCommunityChange={(community) => {
+                changeSelectedCommunity(community);
+              }}
+              chosenCommunity={community}
+              trigger={
+                <Button
+                  size="mini"
+                  className="community-header-button ml-5"
+                  color="blue"
+                  name="setting"
+                  type="button"
+                >
+                  {i18next.t('Edit')}
+                </Button>
+              }
+            />
+          </div>
+        </Container>
       </Container>
-    </Container>
-  );
-};
+    );
+  }
+}
 
 CommunityHeaderComponent.propTypes = {
   imagePlaceholderLink: PropTypes.string.isRequired,
-  community: PropTypes.object.isRequired,
-  setCommunity: PropTypes.func.isRequired,
+  community: PropTypes.shape({
+    uuid: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    logo: PropTypes.string.isRequired,
+  }),
```

Comment:
Silly question: Where is this data filled in?
Community doesn't have title outside the metadata and I see the following definiton below ([code here](https://github.com/inveniosoftware/react-invenio-deposit/pull/451/files#diff-48a39cc50930f6681605ecbd7bf7f66430fc8d3dde2d16292ec629476c25dce1R60-R73)):
```
  if (_selectedCommunity) {
    _selectedCommunity = {
      id: _selectedCommunity.id,
      uuid: _selectedCommunity.uuid || _selectedCommunity,
      metadata: {
        title: _selectedCommunity.title,
        description: _selectedCommunity.description,
        type: 'Type',
      },
      links: {
        self_html: '/',
      },
    };
  }
```